platform/linux-generic/odp_packet_io.c

```diff
@@ -894,102 +864,19 @@ static inline int pktin_recv_buf(pktio_entry_t *entry, int pktin_index,
 		pool = entry->in_queue[pktin_index].vector.pool;
 	}
 
-	num_rx = 0;
-	num_dst = 0;
-	num_ev = 0;
-
-	/* Some compilers need this dummy initialization */
-	cur_queue = ODP_QUEUE_INVALID;
-
-	pkts = entry->ops->recv(entry, pktin_index, packets, num);
+	num_rx = entry->ops->recv(entry, pktin_index, (odp_packet_t *)event_hdrs, num);
 
-	for (i = 0; i < pkts; i++) {
-		pkt = packets[i];
-		pkt_hdr = packet_hdr(pkt);
-		event_hdr = packet_to_event_hdr(pkt);
+	if (!vector_enabled || num_rx < 2)
+		return num_rx;
 
-		if (odp_unlikely(pkt_hdr->p.input_flags.dst_queue)) {
-			/* Sort events for enqueue multi operation(s) based on CoS
-			 * and destination queue. */
-			if (odp_unlikely(num_dst == 0)) {
-				num_dst = 1;
-				cur_queue = pkt_hdr->dst_queue;
-				cur_cos = pkt_hdr->cos;
-				dst[0] = cur_queue;
-				cos[0] = cur_cos;
-				dst_idx[0] = 0;
-			}
-
-			ev[num_ev] = odp_packet_to_event(pkt);
-
-			if (cur_queue != pkt_hdr->dst_queue || cur_cos != pkt_hdr->cos) {
-				cur_queue = pkt_hdr->dst_queue;
-				cur_cos = pkt_hdr->cos;
-				dst[num_dst] = cur_queue;
-				cos[num_dst] = cur_cos;
-				dst_idx[num_dst] = num_ev;
-				num_dst++;
-			}
+	/* Create packet vector */
```

Comment:
The below function name is descriptive so this is slightly unnecessary.

Reply:
Removed in V2.